src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -0,0 +1,129 @@
+        return res['resources'] if 'resources' in res else res
+
+    """
+    @CreatePermission
+    def create(
+        self,
+        cluster_id: str,
+        auth_mode: AuthMode,
+        intent: Intent = Intent.PRESENT,
+        domain_settings: Dict = {},
+        user_group_settings: List[str] = [],
+        custom_dns: List[str] = [],
+        placement: str = '',
+        clustering: Union[SMBClustering, None] = None,
+        public_addrs: List[str] = [],
+        custom_smb_global_options: Dict = {}
+    ) -> results.Simplified:
+        # pylint: disable=unused-argument
+        cluster_data = locals().items()
+
+        resource = {}
+        resource['resource_type'] = self._resource
+
+        for key, value in cluster_data:
+            if key != 'self' and value:
+                if key == 'domain_settings' and domain_settings:
+                    if 'domain_settings' not in resource:
+                        resource[key] = {}
+                        resource[key]['join_sources'] = []
+
+                    resource[key]['realm'] = domain_settings['realm'] if 'realm' in domain_settings else None
+
+                    for auth_id in domain_settings['join_sources'] if 'join_sources' in domain_settings else []:
+                        value = {'source_type': 'resource', 'ref': auth_id}
+                        resource[key]['join_sources'].append(value)
+
+                elif key == 'user_group_settings' and user_group_settings:
+                    if 'user_group_settings' not in resource:
+                            resource[key] = []
+
+                    for user_id in user_group_settings:
+                        value = {'source_type': 'resource', 'ref': user_id}
+                        resource[key].append(value)
+                else:
+                    resource[key] = value
+        res =  mgr.remote('smb', 'apply_resources', json.dumps(resource))
+        return res.to_simplified() if type(res) is results.Result else res
+    """
+
+    @CreatePermission
+    def create(
+        self,
+        cluster_data: Dict
+    ) -> results.Simplified:
+        """
+        Create an smb cluster
+
+        :param cluster_data: Dict containing cluster information
+        :return: Returns cluster resource.
+        :rtype: results.Simplified
+        """
+        return mgr.remote('smb', 'apply_resources', json.dumps(cluster_data)).to_simplified()
+
+
+@APIRouter('/smb/share', Scope.SMB)
+@APIDoc("SMB Share Management API", "SMB")
+class SMBShare(RESTController):
+    _resource = 'ceph.smb.share'
+
+    @ReadPermission
+    def list(self, cluster_id: str = '') -> List[Dict]:
+        """
+        List all smb shares or all shares for a given cluster
+
+        :param cluster_id: Dict containing cluster information
+        :return: Returns list of shares.
+        :rtype: List[Dict]
+        """
+        res = mgr.remote('smb', 'show', [f'{self._resource}.{cluster_id}' if cluster_id else f'{self._resource}'])
```

Comment:
should we have a different `get` endpoint to get the individual resources?

```
mgr.remote('smb', 'show', [f'{self._resource}.{cluster_id}' if cluster_id else f'{self._resource}'])
```

and maybe the above code can exist in an smb service file which?

I think we might be using `smb show` and `smb apply_resources` most of the time so might be good to have it in a single place and consume from there.

Reply:
Okay sure, we can do that. I'll update it
I wasn't sure if there was a need to get them individually